src/AlgebraicGeometry/Schemes/AffineAlgebraicSet/Objects/Methods.jl

```diff
@@ -66,3 +66,83 @@ function Base.show(io::IO, X::AffineAlgebraicSet)
   end
 end
 
+########################################################
+# (2) give back k-points (if finitely many)
+########################################################
+@doc raw"""
+    rational_point_coordinates(I::MPolyIdeal)
+
+Returns the $k$-coordinates of the point corresponding to a 
+maximal ideal $I \in k[x_1,\dots,x_n]$. 
+If $I$ is not maximal or does not describe a point with coordinates in the 
+field $k$, an error exception results.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = QQ[:x, :y]
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> I = ideal(R, [x-1,y-3])
+Ideal generated by
+  x - 1
+  y - 3
+
+julia> rational_point_coordinates(I)
+2-element Vector{QQFieldElem}:
+ 1
+ 3
+
+```
+"""
+function rational_point_coordinates(I::MPolyIdeal)
+  R = base_ring(I)
+  o = degrevlex(gens(R))
+  LG = leading_ideal(I;ordering=o)
+  @req dim(LG)==0 "Ideal does not describe finite set of points"
+  vd = vector_space_dim(quo(base_ring(LG),LG)[1])
+  @req vd ==1 "Ideal does not describe a single K-point"
+  nf_vec = [normal_form(v,I) for v in gens(R)]
+  return [iszero(a) ? zero(coefficient_ring(a)) : leading_coefficient(a) for a in nf_vec]
+end
+
+@doc raw"""
+    rational_points(X::AffineAlgebraicSet)
+
+Returns a vector of coordinate vectors of the $k$-points of $X$ under 
+the condition that $X$ is of dimension zero, where $k$ denotes the base 
+field.
+If $X$ is not zero-dimensional (considered as an algebraic set over the 
+algebraic closure of $k$), an exception is raised.
+
+!!! note
+    If $X$ is zero dimensional, but does not contain any $k$-point,
+the returned vector will be empty.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = QQ[:x, :y]
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> I = ideal(R, [x^2-1,y-3])
+Ideal generated by
+  x^2 - 1
+  y - 3
+
+julia> X = algebraic_set(I)
+Affine algebraic set
+  in affine 2-space over QQ with coordinates [x, y]
+defined by ideal (x^2 - 1, y - 3)
+
+julia> rational_points(X)
+2-element Vector{Vector{QQFieldElem}}:
+ [1, 3]
+ [-1, 3]
+
+```
+"""
+function rational_points(X::AffineAlgebraicSet{T}) where T <: Field 
+  I = fat_ideal(X)
+  @req dim(I) == 0 "Not a zero-dimensional algebraic set"
+  PL = minimal_primes(I)
+  return [rational_point_coordinates(J) for J in PL if vector_space_dim(quo(base_ring(J),J)[1]) == 1]
```

Comment:
Shouldn't the resulting vector be of type `Vector{elem_type(coefficient_ring(base_ring(I)))}` if empty?